test/tests/functional/pbs_multi_sched.py

```diff
@@ -1637,3 +1637,103 @@ def test_dedicated_time_multisched(self):
         self.server.expect(JOB, {ATTR_comment: "Not Running: " + msg}, id=jid2)
         self.scheds['sc2'].log_match(jid2 + ";Dedicated Time",
                                      starttime=self.server.ctime)
+
+    def test_auto_sched_off_due_to_fds_limit(self):
+        """
+        Test to make sure scheduling should be turned off automatically
+        when number of open files per process are exhausted
+        """
+
+        # get the number of open files per process
+        ret = self.du.run_cmd(
+            self.server.hostname, [
+                'ulimit', '-n'], as_script=True, logerr=False)
+        self.assertEqual(ret['rc'], 0)
+        open_files_limit = ret['out'][0]
```

Comment:
There's a python module which you should use instead of this.  It's called resource.  You want resource.RLIMIT_NOFILE

Reply:
Thanks for your suggestion. I am now using resource module of python. I initially observed that we are directly using "ulimit -n" command in some other part of the PTL(pbs_hook_debug_nocrash.py) and hence used run_cmd earlier. May be we need to correct that as well. Also with run_cmd we can execute ulimit -n as root, but with resource module we can not do it like that and hence at the start of the test case I have used the following.

      if os.getuid() != 0 or sys.platform in ('cygwin', 'win32'):
            self.skipTest("Test need to run as root")

Hope it is fine.